ast/ast_test.go

```diff
@@ -1,11 +1,10 @@
-package ast_test
+package ast
```

Comment:
I'm not sold on the merits of using separate testing packages. The advertised benefit is that one should only test exposed interfaces. I think it's ok to write tests for unexported code as well. For example, sometimes there's a complicated helper function that warrants more attention.

Reply:
> sometimes there's a complicated helper function that warrants more attention.

Have you added such a test in this PR? (I quickly scanned over the test but it wasn't clear).

If this change isn't needed for this PR -- I would rather undo this change and leave it for a separate discussion.

Reply:
it's also possible to have multiple test files, some with `package ast_test` and others with `package ast`, which can be used to communicate the goal of the file -- testing exposed vs internal functions.

Reply:
> Have you added such a test in this PR?

No, that was just a general comment. It didn't strike me as necessary to add tests for unexported code in this PR.

I intended to include the package rename change in https://github.com/earthly/earthly/pull/3717, but left it out by accident.

Do you have concerns about the change? It seems like this pattern is only used with the `onpar` tests. The differing package names also confuse my editor.

Reply:
The "_test" package idiom is nice because it allows us to communicate that this test is only for external methods. My concern is that we'll loose the ability to have go enforce this convention.

Reply:
Right, but it's not like it was used extensively here or that removing it introduces further risk. If anything removing it reduces complexity a bit. I just considered this part of the test cleanup & refactoring.

Reply:
my general preference is if a change isn't needed, and the existing code is working, we just leave it as-is, but if you strongly feel that the "_test" package idiom should be avoided then I don't have a strong opinion against the change.

Reply:
We do use the "_test" package idiom in other places too, e.g. https://github.com/earthly/earthly/blob/587f9f694fc093d0c640a6024c93442422d18fa2/util/buildkitskipper/hasher/hasher_test.go#L1

we should come to a decision on if it's something we want to remove everywhere.

Reply:
> change isn't needed, and the existing code is working, we just leave it as-is

Many changes aren't strictly needed but may simplify or improve. 

> remove everywhere

I don't agree that the same approach needs to be used everywhere. The context is important. 

I'll remove this change for now as it's now a bit of a distraction. My original intention was to tidy up and simplify the tests.